google-cloud-firestore/src/main/java/com/google/cloud/firestore/CollectionReference.java

```diff
@@ -48,7 +52,7 @@
    * @param firestore The Firestore client.
    * @param collectionPath The Path of this collection.
    */
-  protected CollectionReference(FirestoreImpl firestore, ResourcePath collectionPath) {
+  CollectionReference(FirestoreRpcContext<?> firestore, ResourcePath collectionPath) {
```

Comment:
Should the constructor be marked `@InternalExtensionOnly` rather than the whole class?

Reply:
Each of the model classes has the following note on it:

> <b>Subclassing Note</b>: Firestore classes are not meant to be subclassed except for use in
> test mocks. Subclassing is not supported in production code and new SDK releases may break code
> that does so.

Given that fact I've gone ahead and added the annotation to each of the classes touched here.

Reply:
Sounds good. I actually did my research this time:

```
Interfaces marked with @InternalExtensionOnly are public, but should only be implemented by internal classes. For the purposes of semver, we reserve the right to add to these interfaces without default implementations (for Java 7).
```

So yes, this should be on the class and not the constructor.